src/main/java/io/logz/jmx2graphite/GraphiteClient.java

```diff
@@ -43,9 +46,9 @@ public GraphiteClient(String serviceHost, String serviceName, String graphiteHos
         logger.info("Graphite Client: using writeTimeoutMs of {} [ms]. Establishing connection..." ,writeTimeoutMs);
 
         SocketFactory socketFactory = new SocketFactoryWithTimeouts(connectTimeout, socketTimeout);
-        if ("udp".equals(protocol)) {
+        if (protocol == UDP) {
             graphite = new GraphiteUDP(new InetSocketAddress(graphiteHostname, graphitePort));
-        } else if ("tcp".equals(protocol)) {
+        } else if (protocol == TCP) {
```

Comment:
Great. 
I would make the last else to be exact match for PICKLED and leave the default else to throw an error, just to be explicit.